docs/design/AnoncredsW3CCompatibility.md

```diff
@@ -0,0 +1,342 @@
+
+In a similar manner, we will proceed through the following processes in comparison with the legacy anoncreds implementations while watching out for signature differences between the two.<br>
+
+- W3C Credential Create
+
+**NOTE: There has been some changes to _encoding of attribute values_ for creating a credential, so we have to be adjust to the new changes.**
+
+```
+async def create_credential(
+        self,
+        credential_offer: dict,
+        credential_request: dict,
+        credential_values: dict,
+    ) -> str:
+...
+...
+  try:
+    credential = await asyncio.get_event_loop().run_in_executor(
+        None,
+        lambda: W3CCredential.create(
+            cred_def.raw_value,
+            cred_def_private.raw_value,
+            credential_offer,
+            credential_request,
+            raw_values,
+            None,
+            None,
+            None,
+            None,
+        ),
+    )
+...
+```
+
+- W3C Credential Request
+
+```
+async def create_w3c_credential_request(
+        self, credential_offer: dict, credential_definition: CredDef, holder_did: str
+    ) -> Tuple[str, str]:
+...
+...
+try:
+  secret = await self.get_master_secret()
+  (
+      cred_req,
+      cred_req_metadata,
+  ) = await asyncio.get_event_loop().run_in_executor(
+      None,
+      W3CCredentialRequest.create,
+      None,
+      holder_did,
+      credential_definition.to_native(),
+      secret,
+      AnonCredsHolder.MASTER_SECRET_ID,
+      credential_offer,
+  )
+...
+```
+
+- W3C Credential Present
+
+```
+async def create_w3c_presentation(
+        self,
+        presentation_request: dict,
+        requested_credentials: dict,
+        schemas: Dict[str, AnonCredsSchema],
+        credential_definitions: Dict[str, CredDef],
+        rev_states: dict = None,
+    ) -> str:
+...
+...
+  try:
+    secret = await self.get_master_secret()
+    presentation = await asyncio.get_event_loop().run_in_executor(
+        None,
+        Presentation.create,
+        presentation_request,
+        present_creds,
+        self_attest,
+        secret,
+        {
+            schema_id: schema.to_native()
+            for schema_id, schema in schemas.items()
+        },
+        {
+            cred_def_id: cred_def.to_native()
+            for cred_def_id, cred_def in credential_definitions.items()
+        },
+    )
+...
+```
+
+#### Converting an already issued legacy anoncreds to W3C format(vice versa)
+
+In this case, we can use `to_w3c` method of `Credential` class to convert from legacy to w3c and `to_legacy` method of `W3CCredential` class to convert from w3c to legacy.<br>
+
+We could call `to_w3c` method like this:
+
+```
+w3c_cred = Credential.to_w3c(cred_def)
+```
+
+and for `to_legacy`:
+
+```
+legacy_cred = W3CCredential.to_legacy()
+```
+
+We don't need to input any parameters to it as it in turn calls `Credential.from_w3c()` method under the hood
+
+### Format Handler for Issue_credential V2_0 Protocol
```

Comment:
@swcurran @dhh1128 @gvelez17 i just added this and i would like your reviews on it. like is this a correct approach to add the w3c format support on the same level as anoncreds?

Reply:
I think this is the correct approach (adding a new formathandler)